File: compiler/testData/loadJava/compiledJava/kotlinSignature/propagation/typeParameter/UseParameterInUpperBound.java
package test;

import java.util.List;

public interface UseParameterInUpperBound {

    public interface Super {
        <A, B extends List<A>> void foo(A a, B b);
    }

    public interface Sub extends Super {
        <B, A extends List<B>> void foo(B b, A a);
    }
}

File: compiler/testData/loadJava/compiledKotlin/fromLoadJava/kotlinSignature/propagation/typeParameter/UseParameterInUpperBound.kt
package test

public interface UseParameterInUpperBound {

    public interface Super {
        public fun <A, B: List<A>> foo(a: A, b: B)
    }

    public interface Sub: Super {
        override fun <B, A: List<B>> foo(a: B, b: A)
    }
}
